src/main/java/uk/ac/ebi/ampt2d/metadata/persistence/entities/Sample.java

```diff
@@ -70,4 +76,18 @@ public Long getId() {
     public AccessionVersionId getAccessionVersionId() {
         return accessionVersionId;
     }
+
+    @PrePersist
+    public void validateTaxonomiesPersist() {
+        if (this.taxonomies == null || this.taxonomies.size() == 0 || this.taxonomies.get(0) == null) {
+            throw new IllegalArgumentException ("Please provide Taxonomies; it can not be null or blank");
+        }
+    }
+
+    @PreUpdate
+    public void validateTaxonomiesUpdate() {
```

Comment:
You could add the annotation to the same function.

Reply:
well, at this moment both are same. if there is any change for updates in future... i am keeping theme separate. Let me know